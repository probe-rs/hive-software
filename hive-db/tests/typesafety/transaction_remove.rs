use hive_db::{CborTransactional, HiveDb, Key};

const FLUSH_INTERVAL_MS: u64 = 60_000;
const CACHE_CAPACITY: u64 = 52_428_800;

fn main() {
    let db = HiveDb::open("dummy_path", FLUSH_INTERVAL_MS, CACHE_CAPACITY);

    let tree = db.open_tree("tree");

    let my_key: Key<u8> = Key::new("NiceNumber");

    tree.transaction(|tree| {
        let previous: Option<bool> = tree.c_remove(&my_key)?;

        Ok(())
    })
    .unwrap();
}
