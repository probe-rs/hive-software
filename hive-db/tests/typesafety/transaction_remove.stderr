error[E0308]: `?` operator has incompatible types
  --> tests/typesafety/transaction_remove.rs:14:38
   |
14 |         let previous: Option<bool> = tree.b_remove(&my_key)?;
   |                                      ^^^^^^^^^^^^^^^^^^^^^^^ expected `Option<bool>`, found `Option<u8>`
   |
   = note: `?` operator cannot convert from `Option<u8>` to `Option<bool>`
   = note: expected enum `Option<bool>`
              found enum `Option<u8>`
